src/chainparams.cpp

```diff
@@ -346,28 +346,76 @@ class CRegTestParams : public CChainParams {
     }
 };
 
+/**
+ * Custom params for testing.
+ */
+class CCustomParams : public CRegTestParams {
+
+    void UpdateFromArgs(ArgsManager& args)
+    {
+        consensus.nSubsidyHalvingInterval = args.GetArg("-con_nsubsidyhalvinginterval", consensus.nSubsidyHalvingInterval);
+        consensus.BIP16Height = args.GetArg("-con_bip16height", consensus.BIP16Height);
+        consensus.BIP34Height = args.GetArg("-con_bip34height", consensus.BIP34Height);
+        consensus.BIP34Hash = uint256S(args.GetArg("-con_bip34hash", "0x0"));
+        consensus.BIP65Height = args.GetArg("-con_bip65height", consensus.BIP65Height);
+        consensus.BIP66Height = args.GetArg("-con_bip66height", consensus.BIP66Height);
+        consensus.powLimit = uint256S(args.GetArg("-con_powlimit", "7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"));
+        consensus.nPowTargetTimespan = args.GetArg("-con_npowtargettimespan", consensus.nPowTargetTimespan);
+        consensus.nPowTargetSpacing = args.GetArg("-con_npowtargetspacing", consensus.nPowTargetSpacing);
+        consensus.fPowAllowMinDifficultyBlocks = args.GetBoolArg("-con_fpowallowmindifficultyblocks", consensus.fPowAllowMinDifficultyBlocks);
+        consensus.fPowNoRetargeting = args.GetBoolArg("-con_fpownoretargeting", consensus.fPowNoRetargeting);
+        consensus.nRuleChangeActivationThreshold = args.GetArg("-con_nrulechangeactivationthreshold", consensus.nRuleChangeActivationThreshold);
+        consensus.nMinerConfirmationWindow = args.GetArg("-con_nminerconfirmationwindow", consensus.nMinerConfirmationWindow);
+
+        consensus.nMinimumChainWork = uint256S(args.GetArg("-con_nminimumchainwork", "0x0"));
+        consensus.defaultAssumeValid = uint256S(args.GetArg("-con_defaultassumevalid", "0x00"));
+
+        nDefaultPort = args.GetArg("-ndefaultport", nDefaultPort);
```

Comment:
Redundant given `port=`?

Reply:
yes, it is redundant, but this way people creating new chains will remember to set it (or set port).